model/trigger/host_spawn.go

```diff
@@ -0,0 +1,194 @@
+package trigger
+
+import (
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/model/notification"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	registry.registerEventHandler(event.ResourceTypeHost, event.EventHostProvisioned, makeSpawnHostTriggers)
+	registry.registerEventHandler(event.ResourceTypeHost, event.EventHostProvisionFailed, makeSpawnHostTriggers)
+}
+
+type spawnHostTriggers struct {
```

Comment:
These triggers and the host expiration trigger should really be the same group. How come you're splitting them up?

Reply:
Because eventHandlers are mapped from `(resource_type, event_type)` to eventHandler. Host expiration is `(event.ResourceTypeHost, event.EventHostExpirationWarningSent)`, while this one is `(event.ResourceTypeHost, event.EventHostProvisioned)` and `(event.ResourceTypeHost, event.EventHostProvisionFailed)`.

Reply:
Yes, but all those events should map to the same set of triggers. They have the same selectors and are logically the same, just have different payloads

Reply:
Let's discuss this out of band of this PR, because what you're proposing requires refactoring all triggers